### PR TITLE
feat: hide shorts comments button

### PR DESCRIPTION
**Pull request by OxrxL**
Immagine 2022-10-26 103256

https://github.com/revanced/revanced-integrations/pull/188

### CODE REVIEW

**Comment by johnconner122** [comment]
Isn't it possible to enable dislike count on shorts?

**Comment by OxrxL** [comment]
> Isn't it possible to enable dislike count on shorts?

1) I don't know.
2) I don't care.
3) This is OT.

File: src/main/kotlin/app/revanced/patches/youtube/layout/comments/patch/CommentsPatch.kt [hunk 1]
```diff
@@ -44,10 +56,48 @@ class CommentsPatch : ResourcePatch {
...
+            checkCastAnchorIndex + 1, """
+                invoke-static {v${(shortsCommentsButtonMethod.instruction(checkCastAnchorIndex) as OneRegisterInstruction).registerA}}, Lapp/revanced/integrations/patches/HideShortsCommentsButtonPatch;->hideShortsCommentsButton(Landroid/view/View;)V
+            """
+        )
+
         return PatchResultSuccess()
     }
+
+    internal companion object {
+        internal var shortsCommentsButtonId: Long = ResourceMappingResourcePatch.resourceMappings.single {
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/comments/patch/CommentsPatch.kt:99]: This is unsafe, an explanation is available in another patch with similar usage.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/layout/comments/patch/CommentsPatch.kt (hunk 1 above)
Comment (OxrxL) [reply src/main/kotlin/app/revanced/patches/youtube/layout/comments/patch/CommentsPatch.kt]: Done! I took the sponsorblock fingerprint as an example.

File: src/main/kotlin/app/revanced/patches/youtube/layout/comments/bytecode/patch/CommentsPatch.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                Opcode.CONST,
+                Opcode.CONST_HIGH16,
+                Opcode.IF_EQZ,
+                Opcode.CONST,
+                Opcode.INVOKE_STATIC,
+                Opcode.MOVE_RESULT_OBJECT,
+                Opcode.CHECK_CAST,
+            )
+        ) {}
+        val checkCastAnchorIndex = checkCastAnchorFingerprint.also {
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/comments/bytecode/patch/CommentsPatch.kt:55]:
```suggestion

        val checkCastAnchorIndex = checkCastAnchorFingerprint.also {
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/comments/patch/CommentsPatch.kt (hunk 1 above)
Comment (OxrxL) [reply src/main/kotlin/app/revanced/patches/youtube/layout/comments/patch/CommentsPatch.kt]: Done! I took the ThemeSetterSystem fingerprint as example.

File: src/main/kotlin/app/revanced/patches/youtube/layout/comments/bytecode/patch/CommentsPatch.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.PreferenceScreen
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.instruction.OneRegisterInstruction
+
+@Patch
+@DependsOn([IntegrationsPatch::class, CommentsResourcePatch::class])
+@Name("comments")
+@Description("Hides comments components below the video player.")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/comments/bytecode/patch/CommentsPatch.kt:32]:
```suggestion
@Description("Hides components related to comments.")
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Pull request merged**